<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Sample Size and Power Analysis</title>

<script src="site_libs/header-attrs-2.8/header-attrs.js"></script>
<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/kePrint-0.0.1/kePrint.js"></script>
<link href="site_libs/lightable-0.0.1/lightable.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />
<script>
function toggle_visibility(id1, id2) {
var e = document.getElementById(id1);
var f = document.getElementById(id2);
e.style.display = ((e.style.display!='none') ? 'none' : 'block');
if(f.classList.contains('fa-plus')) {
    f.classList.add('fa-minus')
    f.classList.remove('fa-plus')
} else {
    f.classList.add('fa-plus')
    f.classList.remove('fa-minus')
}
}
</script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>





<link rel="stylesheet" href="assets/style-labs.css" type="text/css" />



<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"><strong>DAPR2</strong></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">
    <span class="fas fa-home"></span>
     
  </a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Simple Linear Regression
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="01_models.html">1/1: Functions and models</a>
    </li>
    <li>
      <a href="02_slr.html">1/2: Simple linear regression</a>
    </li>
    <li>
      <a href="03_slr_model_fit.html">1/3: Model Fit, Standardized Coefficients</a>
    </li>
    <li>
      <a href="04_slr_assumptions.html">1/4: Assumptions &amp; Diagnostics</a>
    </li>
    <li>
      <a href="05_slr_writeup.html">1/5: Writing-up</a>
    </li>
    <li class="dropdown-header">1/6: Break week</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Multiple Linear Regression
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="06_mlr.html">1/7: Basics</a>
    </li>
    <li>
      <a href="07_mlr_int.html">1/8: Interactions</a>
    </li>
    <li>
      <a href="08_mlr_assumpt.html">1/9: Assumptions &amp; Diagnostics</a>
    </li>
    <li>
      <a href="09_mlr_select.html">1/10: Model Fit, Model Comparison, Model Selection</a>
    </li>
    <li>
      <a href="10_mlr_report.html">1/11: Writing-up</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Experimental Data
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="11_anova.html">2/1: ANOVA</a>
    </li>
    <li>
      <a href="11_anova.html">2/2: ANOVA (again)</a>
    </li>
    <li>
      <a href="12_factorial_anova.html">2/3: 2x2 ANOVA</a>
    </li>
    <li>
      <a href="13_multiplecomp.html">2/4: Multiple Comparisons</a>
    </li>
    <li class="dropdown-header">2/5: No exercises</li>
    <li class="dropdown-header">2/6: Break week</li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Advanced Topics for LM
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="14_bootstrap_reg.html">2/7: Bootstrap</a>
    </li>
    <li>
      <a href="15_binary_logistic.html">2/8: Binary Logistic Regression</a>
    </li>
    <li>
      <a href="16_more_about_logistic.html">2/9: More About Logistic Regression</a>
    </li>
    <li>
      <a href="17_power_regression.html">2/10: Sample Size and Power Analysis</a>
    </li>
  </ul>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Help
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="intro_r_rstudio.html">Getting started with R &amp; RStudio</a>
    </li>
  </ul>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Sample Size and Power Analysis</h1>

</div>


<p>Before describing and analysing data, you have to collect data. If you’re lucky enough, someone else will have collected the data you need and these will be provided to you, but other times you have to collect the data needed to answer your research question yourself. This leads to some experimental design issues, such as how big should the sample be?</p>
<p>This is exactly the question that we will be answering this week discussing power analysis.</p>
<p>Power analysis lets you determine how many subjects are needed for your study or, conversely, whether your study is any worth pursuing if you can only afford to sample <span class="math inline">\(n\)</span> units.</p>
<div class="frame">
<p><strong>Notation</strong></p>
<p>In the following, the symbol <span class="math inline">\(P()\)</span> denotes the probability of whatever is within the parentheses.</p>
</div>
<div id="errors-and-power-in-hypothesis-testing" class="section level1">
<h1>Errors and Power in Hypothesis Testing</h1>
<p>When testing an hypothesis, we reach one of the following two decisions:</p>
<ul>
<li>not rejecting <span class="math inline">\(H_0\)</span> as the evidence against it is not sufficient</li>
<li>rejecting <span class="math inline">\(H_0\)</span> as we have enough evidence against it</li>
</ul>
<p>However, irrespectively of our decision, the underlying truth can either be that</p>
<ul>
<li><span class="math inline">\(H_0\)</span> is actually false</li>
<li><span class="math inline">\(H_0\)</span> is actually true</li>
</ul>
<p>Hence, we have four possible outcomes following an hypothesis test:</p>
<ol style="list-style-type: decimal">
<li>We did not reject <span class="math inline">\(H_0\)</span> when it is in fact true, meaning we made a <strong>Correct</strong> decision</li>
<li>We rejected <span class="math inline">\(H_0\)</span> when it is in fact false, meaning we made a <strong>Correct</strong> decision</li>
<li>We rejected <span class="math inline">\(H_0\)</span> when it is in fact true, committing a <strong>Type I error</strong> (pronounced “type 1 error”)</li>
<li>We did not reject <span class="math inline">\(H_0\)</span> when it is in fact false, committing a <strong>Type II error</strong> (pronounced “type 2 error”)</li>
</ol>
<p>In the first two cases we are correct, while in the latter two we committed an error.</p>
<p>When we reject the null hypothesis, we never know if we were correct or committed a Type I error, however we can control the chance of us committing a Type I error.
Similarly, when we do not reject the null hypothesis, we never know if we are correct or we committed a Type II error, but we can also control the chance of us committing a Type II error.</p>
<p>The following table summarises the two types of errors that we can commit:</p>
<p><img src="images/ht-errors-table-2.png" width="80%" style="display: block; margin: auto;" /></p>
<p>A Type I error corresponds to a false discovery, while a type II error corresponds to a failed discovery/missed opportunity.</p>
<p>Each error has a corresponding probability:</p>
<ul>
<li>The probability of incorrectly rejecting a true null hypothesis is <span class="math inline">\(\alpha = P(\text{Type I error})\)</span></li>
<li>The probability of incorrectly not rejecting a false null hypothesis is <span class="math inline">\(\beta = P(\text{Type II error})\)</span></li>
</ul>
<p>A related quantity is <strong>Power</strong>, which is defined as the probability of correctly rejecting a false null hypothesis.</p>
<div class="yellow">
<p><strong>Power</strong></p>
<p>Power is the probability of rejecting a false null hypothesis. That is, it is the probability that we will find an effect when it is in fact present.
<span class="math display">\[
\text{Power} = 1 - P(\text{Type II error}) = 1 - \beta
\]</span></p>
</div>
</div>
<div id="factors-affecting-power" class="section level1">
<h1>Factors affecting power</h1>
<p>In practice, it is ideal for studies to have high power while using a relatively small significance level such as 0.05 or 0.01. For a fixed <span class="math inline">\(\alpha\)</span>, the power increases in the same cases that P(Type II error) decreases, namely as the sample size increases and as the parameter value moves farther into the H values away from
the H0 value.</p>
<p>The <strong>power</strong> of a test is affected by the following factors:</p>
<ul>
<li><strong>sample size.</strong> Power increases as the sample size increases.</li>
<li><strong>effect size.</strong> Power increases as the parameter value moves farther into the <span class="math inline">\(H_1\)</span> values away from the <span class="math inline">\(H_0\)</span> value.</li>
<li><strong>significance level.</strong> Power increases as the significance level increases.</li>
</ul>
<p>Out of these, increasing the significance level <span class="math inline">\(\alpha\)</span> is never an acceptable way to increase power as it leads to more Type I errors, i.e. a higher chance of incorrectly rejecting a true null hypothesis (false discoveries).</p>
<p>As you can see the four quantities above are linked in some complex way, and in this exercises you will see:</p>
<ol style="list-style-type: lower-alpha">
<li><p>How to compute the minimum sample size required to correctly reject a false null hypothesis with a given degree of confidence.</p></li>
<li><p>Given sample size constraints, to find what is the probability that you test will correctly reject a false null hypothesis. If it’s too low, you might want to reconsider your study or perhaps even abandon it.</p></li>
</ol>
<div class="frame">
<p><strong>Did You Know?</strong></p>
<p>Before granting research funds, many funding agencies expect researchers to show that for the planned study, reasonable power (usually at least 0.80) exists at values of the parameter that are considered practically significant.</p>
</div>
</div>
<div id="effect-size" class="section level1">
<h1>Effect size</h1>
<p>Effect size refers to the “detectability” of your alternative hypothesis. In simple terms, it compares the distance between the alternative and the null hypothesis to the variability in your data.</p>
<p>For simplicity consider comparing a mean: <span class="math inline">\(H_0: \mu = 0\)</span> vs <span class="math inline">\(H_1: \mu \neq 0\)</span>. If the sample mean were really 0.1 and the null hypothesis is 0, the distance is 0.1 - 0 = 0.1.</p>
<p>Now, a distance of 0.1 has a different weight in the following two scenarios.</p>
<p><strong>Scenario 1.</strong> Data vary between -1 and 1.</p>
<p><strong>Scenario 2.</strong> Data vary between -1000 and 1000.</p>
<p>Clearly, in Scenario 1 a distance of of 0.1 is a big difference.
Conversely, in Scenario 2 a distance of 0.1 is not an interesting difference, it’s negligible compared to the magnitude of the data.</p>
<p><img src="17_power_regression_files/figure-html/unnamed-chunk-2-1.png" width="85%" style="display: block; margin: auto;" /></p>
</div>
<div id="the-pwr-package" class="section level1">
<h1>The pwr package</h1>
<p>We will perform power analysis using the <code>pwr</code> package. To install it, run <code>install.packages("pwr")</code>.</p>
<p>The following functions are available.</p>
<table>
<thead>
<tr class="header">
<th align="left">Function</th>
<th align="left">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left"><code>pwr.2p.test</code></td>
<td align="left">Two proportions (equal n)</td>
</tr>
<tr class="even">
<td align="left"><code>pwr.2p2n.test</code></td>
<td align="left">Two proportions (unequal n)</td>
</tr>
<tr class="odd">
<td align="left"><code>pwr.anova.test</code></td>
<td align="left">Balanced one-way ANOVA</td>
</tr>
<tr class="even">
<td align="left"><code>pwr.chisq.test</code></td>
<td align="left">Chi-square test</td>
</tr>
<tr class="odd">
<td align="left"><code>pwr.f2.test</code></td>
<td align="left">General linear model</td>
</tr>
<tr class="even">
<td align="left"><code>pwr.p.test</code></td>
<td align="left">Proportion (one sample)</td>
</tr>
<tr class="odd">
<td align="left"><code>pwr.r.test</code></td>
<td align="left">Correlation</td>
</tr>
<tr class="even">
<td align="left"><code>pwr.t.test</code></td>
<td align="left">t-tests (one sample, two samples, paired)</td>
</tr>
<tr class="odd">
<td align="left"><code>pwr.t2n.test</code></td>
<td align="left">t-test (two samples with unequal n)</td>
</tr>
</tbody>
</table>
<p>For each function, you can specify three of four arguments (sample size, alpha, effect size, power) and the fourth argument will be calculated for you.</p>
<p>Of the four quantities, effect size is often the most difficult to specify. Calculating effect size typically requires some experience with the measures involved and knowledge of past research.</p>
<p>Typically, specifying effect size requires you to read published literature or past papers on your research topic, to see what effect sizes were found and what significant results were reported. Other times, this might come from previous collected data or subject-knowledge from your colleagues.</p>
<p>But what can you do if you have no clue what effect size to expect in a given study? Cohen (1988)<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> provided guidelines for what a small, medium, or large effect typically is in the behavioural sciences.</p>
<table>
<thead>
<tr class="header">
<th align="left">Type of test</th>
<th align="center">Small</th>
<th align="center">Medium</th>
<th align="center">Large</th>
<th align="center"></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td align="left">t-test</td>
<td align="center">0.20</td>
<td align="center">0.50</td>
<td align="center">0.80</td>
<td align="center"></td>
</tr>
<tr class="even">
<td align="left">ANOVA</td>
<td align="center">0.10</td>
<td align="center">0.25</td>
<td align="center">0.40</td>
<td align="center"></td>
</tr>
<tr class="odd">
<td align="left">Linear regression</td>
<td align="center">0.02</td>
<td align="center">0.15</td>
<td align="center">0.35</td>
<td align="center"></td>
</tr>
</tbody>
</table>
<p>For more information, please refer to Cohen J. (1992).<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a></p>
</div>
<div id="a.-power-for-t-tests" class="section level1">
<h1>A. Power for t-tests</h1>
<p>We compare the mean of a response variable between two groups using a t-test. For example, if you are comparing the mean response between two groups, say treatment and control, the null and alternative hypotheses are:
<span class="math display">\[
H_0 : \mu_t - \mu_c = 0 \\
H_1 : \mu_t - \mu_c \neq 0
\]</span></p>
<p>The effect size in this case is Cohen’s <span class="math inline">\(D\)</span>:
<span class="math display">\[
D = \frac{(\bar x_t - \bar x_c) - 0}{s_p}
\]</span>
where</p>
<ul>
<li><span class="math inline">\(\bar x_t\)</span> and <span class="math inline">\(\bar x_c\)</span> are the sample means in the treatment and control groups, respectively</li>
<li><span class="math inline">\(s_p\)</span> is the “pooled” standard deviation</li>
</ul>
<p>Cohen’s <span class="math inline">\(D\)</span> measures the distance between (a) the observed difference in means from (b) the hypothesised value 0, and compares this to the variability in the data.</p>
<p>In R we use the function</p>
<pre><code>pwr.t.test(n = , d = , sig.level = , power = , type = , alternative = )</code></pre>
<p>where</p>
<ul>
<li><code>n</code> = the sample size</li>
<li><code>d</code> = the effect size</li>
<li><code>sig.level</code> = the significance level <span class="math inline">\(\alpha\)</span> (the default is 0.05)</li>
<li><code>power</code> = the power level.</li>
<li><code>type</code> = the type of t-test to perform: either a two-sample t-test (“two.sample”), a one-sample t-test (“one.sample”), or a dependent sample t-test (“paired”). A two-sample test is the default.</li>
<li><code>alternative</code> = whether the alternative hypothesis is two-sided (“two.sided”)
or one-sided (“less” or “greater”). A two-sided test is the default.</li>
</ul>
<p>Consider the following example. You are interested in whether the presence of a colour distractor influences reaction time. You measure reaction times for a group of participants each taking either an experiment with a colour distractor or a standard one. You perform a two-sided independent samples t-test for
<span class="math display">\[
H_0 : \mu_c - \mu_s = 0 \\
H_1 : \mu_c - \mu_s \neq 0
\]</span></p>
<p>Suppose that from past experience (or previous literature) you know that reaction times has a standard deviation of roughly 1.4 seconds.
Furthermore, suppose you consider a 1 second difference in reaction times an important difference to be detected.</p>
<p>Your aim is then to be able to detect an effect of size <span class="math inline">\(D = \frac{1}{1.4}=0.7\)</span> or larger.
You also want to be sure to detect such a difference, if it truly exists, 90% of the times, while keeping the chance of a false discovery to 5%.
How many participants will you need to study?</p>
<p>All we need to do is insert this information in the <code>pwr.t.test()</code> function:</p>
<pre class="r"><code>library(pwr)
res_t1 &lt;- pwr.t.test(d = 0.7, sig.level = 0.05, power = 0.90, type = &quot;two.sample&quot;, 
           alternative = &quot;two.sided&quot;)
res_t1</code></pre>
<pre><code>## 
##      Two-sample t test power calculation 
## 
##               n = 43.87041
##               d = 0.7
##       sig.level = 0.05
##           power = 0.9
##     alternative = two.sided
## 
## NOTE: n is number in *each* group</code></pre>
<p>The output above tells us that, in order to detect a standardized mean difference (i.e. an effect size) of <code>d = 0.7</code> with a 90% probability, while limiting the probability of a false discovery to <code>0.05</code> in a two-sided independent sample t-test, you need at least <code>n = 43.87041</code> individuals in each group.</p>
<p>Now, <code>n = 43.87041</code> is not an integer number. You must always round <strong>in excess</strong>, and in this case you might want to have <span class="math inline">\(n = 44\)</span> or <span class="math inline">\(n = 45\)</span> individuals in each group.</p>
<p>You can also plot the output created above to see how the power varies with the sample size:</p>
<pre class="r"><code>plot(res_t1)</code></pre>
<p><img src="17_power_regression_files/figure-html/unnamed-chunk-4-1.png" width="60%" style="display: block; margin: auto;" /></p>
<p><br>
Let’s now slightly alter our perspective. Suppose that we already know that our survey will be limited by budget or time constraints and we can only afford to survey 40 individuals in total — that is, 20 in each group. Furthermore, we want to detect a difference in means of 0.6 standard deviations.
What’s the probability that you’ll be able to detect a difference between the population means that’s this large, given the constraints outlined?</p>
<pre class="r"><code>res_t2 &lt;- pwr.t.test(n = 20, d = 0.6, sig.level = 0.05, 
                     type = &quot;two.sample&quot;, alternative = &quot;two.sided&quot;)
res_t2</code></pre>
<pre><code>## 
##      Two-sample t test power calculation 
## 
##               n = 20
##               d = 0.6
##       sig.level = 0.05
##           power = 0.4560341
##     alternative = two.sided
## 
## NOTE: n is number in *each* group</code></pre>
<p>Affording only 20 participants per group, and only tolerating a 5% chance of a false discovery (i.e. rejecting the null when it is true), we would only be able to detect an effect of size 0.6 with a 45.6% chance, i.e. less than 50%.
Similarly, you have a 50% chance of not finding the effect you’re looking for. The success rate of your test, given your survey constraints, is like flipping a coin! You might want to rethink this experiment, maybe try to improve the data collection or the design, otherwise you might end up wasting time.</p>
<p><br></p>
<p>If you wish to have different sample sizes in the two groups, <code>n1</code> and <code>n2</code>, you should use the function</p>
<pre><code>pwr.t2n.test(n1 = , n2 = , d = , sig.level = , power = , alternative = )</code></pre>
<p>Don’t be afraid of it, and try exploring it by varying its arguments!</p>
</div>
<div id="b.-power-for-linear-regression" class="section level1">
<h1>B. Power for linear regression</h1>
<p>In linear regression, the relevant function is</p>
<pre><code>pwr.f2.test(u = , v = , f2 = , sig.level = , power = )</code></pre>
<p>where</p>
<ul>
<li><code>u</code> = numerator degrees of freedom</li>
<li><code>v</code> = denominator degrees of freedom</li>
<li><code>f2</code> = effect size</li>
</ul>
<p>The formula for the effect size <span class="math inline">\(f^2\)</span> is different depending on your goal.</p>
<div id="goal-1" class="section level3">
<h3>Goal 1</h3>
<p><strong>Goal 1.</strong> Evaluating the impact of a set of predictors on an dependent variable.</p>
<p>Here you have one model,
<span class="math display">\[
y = \beta_0 + \beta_1 x_1 + \dots + \beta_k x_k + \epsilon
\]</span></p>
<p>and you wish to find the minimum sample size required to answer the following test of hypothesis with a given power:
<span class="math display">\[
\begin{aligned}
H_0 &amp;: \beta_1 = \beta_2 = \dots = \beta_k = 0 \\
H_1 &amp;: \text{At least one } \beta_i \neq 0
\end{aligned}
\]</span></p>
<p>The appropriate formula for the effect size is:
<span class="math display">\[
f^2 = \frac{R^2}{1 - R^2}
\]</span></p>
<p>And the numerator degrees of freedom are <span class="math inline">\(\texttt u = k\)</span>, the number of predictors in the model.</p>
<p>The denominator degrees of freedom returned by the function will give you:
<span class="math display">\[
\texttt v = n - (k + 1) = n - k - 1
\]</span></p>
<p>From which you can infer the sample size as
<span class="math display">\[
n = \texttt v + k + 1
\]</span></p>
</div>
<div id="goal-2" class="section level3">
<h3>Goal 2</h3>
<p><strong>Goal 2.</strong> Evaluating the impact of <strong>extra</strong> predictors on an dependent variable, above and beyond other predictors.</p>
<p>In this case you would have a smaller model <span class="math inline">\(m\)</span> with <span class="math inline">\(k\)</span> predictors and a larger model <span class="math inline">\(M\)</span> with <span class="math inline">\(K\)</span> predictors:
<span class="math display">\[
\begin{aligned}
m &amp;: \quad y = \beta_0 + \beta_1 x_1 + \dots + \beta_{k} x_k 
+ \epsilon \\
M &amp;: \quad y = \beta_0 + \beta_1 x_1 + \dots + \beta_{k} x_k 
+ \underbrace{\beta_{k + 1} x_{k+1} + \dots + \beta_{K} x_K}_{\text{extra predictors}}
+ \epsilon
\end{aligned}
\]</span></p>
<p>This case is when you wish to find the minimum sample size required to answer the following test of hypothesis:
<span class="math display">\[
\begin{aligned}
H_0 &amp;: \beta_{k+1} = \beta_{k+2} = \dots = \beta_K = 0 \\
H_1 &amp;: \text{At least one of the above } \beta \neq 0
\end{aligned}
\]</span></p>
<p>You need to use the R-squared from the larger model <span class="math inline">\(R^2_M\)</span> and the R-squared from the smaller model <span class="math inline">\(R^2_m\)</span>. The appropriate formula for the effect size is:
<span class="math display">\[
f^2 = \frac{R^2_{M} - R^2_{m}}{1 - R^2_M}
\]</span></p>
<p>Here, the numerator degrees of freedom are the extra predictors: <span class="math inline">\(\texttt u = K - k\)</span>.</p>
<p>The denominator degrees of freedom returned by the function will give you (here you use <span class="math inline">\(K\)</span> the number of all predictors in the larger model):
<span class="math display">\[
\texttt v = n - (K + 1) = n - K - 1
\]</span></p>
<p>From which you can infer the sample size as
<span class="math display">\[
n = \texttt v + K + 1
\]</span></p>
<p><br></p>
<div class="optional-begin">
Wellbeing/Rurality data codebook. Click the plus to expand →<span id="opt-start-217" class="fa fa-plus optional-icon clickable" onclick="toggle_visibility(&#39;opt-body-217&#39;, &#39;opt-start-217&#39;)"></span>
</div>
<div id="opt-body-217" class="optional-body" style="display: none;">
<p><strong>Download link</strong></p>
<p>The data is available at <a href="https://uoepsy.github.io/data/wellbeing_rural.csv" class="uri">https://uoepsy.github.io/data/wellbeing_rural.csv</a>.</p>
<p><strong>Description</strong></p>
<p>From the Edinburgh &amp; Lothians, 100 city/suburb residences and 100 rural residences were chosen at random and contacted to participate in the study. The Warwick-Edinburgh Mental Wellbeing Scale (WEMWBS), was used to measure mental health and well-being.
Participants filled out a questionnaire including items concerning: estimated average number of hours spent outdoors each week, estimated average number of social interactions each week (whether on-line or in-person), whether a daily routine is followed (yes/no). For those respondents who had an activity tracker app or smart watch, they were asked to provide their average weekly number of steps.</p>
<p>The data in <code>wellbeing_rural.csv</code> contain seven attributes collected from a random sample of <span class="math inline">\(n=200\)</span> hypothetical residents over Edinburgh &amp; Lothians, and include:</p>
<ul>
<li><code>wellbeing</code>: Warwick-Edinburgh Mental Wellbeing Scale (WEMWBS), a self-report measure of mental health and well-being. The scale is scored by summing responses to each item, with items answered on a 1 to 5 Likert scale. The minimum scale score is 14 and the maximum is 70.<br />
</li>
<li><code>outdoor_time</code>: Self report estimated number of hours per week spent outdoors<br />
</li>
<li><code>social_int</code>: Self report estimated number of social interactions per week (both online and in-person)</li>
<li><code>routine</code>: Binary 1=Yes/0=No response to the question “Do you follow a daily routine throughout the week?”</li>
<li><code>location</code>: Location of primary residence (City, Suburb, Rural)</li>
<li><code>steps_k</code>: Average weekly number of steps in thousands (as given by activity tracker if available)</li>
<li><code>age</code>: Age in years of respondent</li>
</ul>
<p><strong>Preview</strong></p>
<p>The first six rows of the data are:</p>
<div id="mnnovkyrpc" style="overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>html {
  font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Helvetica Neue', 'Fira Sans', 'Droid Sans', Arial, sans-serif;
}

#mnnovkyrpc .gt_table {
  display: table;
  border-collapse: collapse;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#mnnovkyrpc .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#mnnovkyrpc .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#mnnovkyrpc .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 0;
  padding-bottom: 4px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#mnnovkyrpc .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#mnnovkyrpc .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#mnnovkyrpc .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#mnnovkyrpc .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#mnnovkyrpc .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#mnnovkyrpc .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#mnnovkyrpc .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#mnnovkyrpc .gt_group_heading {
  padding: 8px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
}

#mnnovkyrpc .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#mnnovkyrpc .gt_from_md > :first-child {
  margin-top: 0;
}

#mnnovkyrpc .gt_from_md > :last-child {
  margin-bottom: 0;
}

#mnnovkyrpc .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#mnnovkyrpc .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 12px;
}

#mnnovkyrpc .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#mnnovkyrpc .gt_first_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
}

#mnnovkyrpc .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#mnnovkyrpc .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#mnnovkyrpc .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#mnnovkyrpc .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#mnnovkyrpc .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#mnnovkyrpc .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding: 4px;
}

#mnnovkyrpc .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#mnnovkyrpc .gt_sourcenote {
  font-size: 90%;
  padding: 4px;
}

#mnnovkyrpc .gt_left {
  text-align: left;
}

#mnnovkyrpc .gt_center {
  text-align: center;
}

#mnnovkyrpc .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#mnnovkyrpc .gt_font_normal {
  font-weight: normal;
}

#mnnovkyrpc .gt_font_bold {
  font-weight: bold;
}

#mnnovkyrpc .gt_font_italic {
  font-style: italic;
}

#mnnovkyrpc .gt_super {
  font-size: 65%;
}

#mnnovkyrpc .gt_footnote_marks {
  font-style: italic;
  font-weight: normal;
  font-size: 65%;
}
</style>
<table class="gt_table">
  
  <thead class="gt_col_headings">
    <tr>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">age</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">outdoor_time</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">social_int</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">routine</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">wellbeing</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1">location</th>
      <th class="gt_col_heading gt_columns_bottom_border gt_right" rowspan="1" colspan="1">steps_k</th>
    </tr>
  </thead>
  <tbody class="gt_table_body">
    <tr><td class="gt_row gt_right">28</td>
<td class="gt_row gt_right">12</td>
<td class="gt_row gt_right">13</td>
<td class="gt_row gt_right">1</td>
<td class="gt_row gt_right">36</td>
<td class="gt_row gt_left">rural</td>
<td class="gt_row gt_right">21.6</td></tr>
    <tr><td class="gt_row gt_right">56</td>
<td class="gt_row gt_right">5</td>
<td class="gt_row gt_right">15</td>
<td class="gt_row gt_right">1</td>
<td class="gt_row gt_right">41</td>
<td class="gt_row gt_left">rural</td>
<td class="gt_row gt_right">12.3</td></tr>
    <tr><td class="gt_row gt_right">25</td>
<td class="gt_row gt_right">19</td>
<td class="gt_row gt_right">11</td>
<td class="gt_row gt_right">1</td>
<td class="gt_row gt_right">35</td>
<td class="gt_row gt_left">rural</td>
<td class="gt_row gt_right">49.8</td></tr>
    <tr><td class="gt_row gt_right">60</td>
<td class="gt_row gt_right">25</td>
<td class="gt_row gt_right">15</td>
<td class="gt_row gt_right">0</td>
<td class="gt_row gt_right">35</td>
<td class="gt_row gt_left">rural</td>
<td class="gt_row gt_right">NA</td></tr>
    <tr><td class="gt_row gt_right">19</td>
<td class="gt_row gt_right">9</td>
<td class="gt_row gt_right">18</td>
<td class="gt_row gt_right">1</td>
<td class="gt_row gt_right">32</td>
<td class="gt_row gt_left">rural</td>
<td class="gt_row gt_right">48.1</td></tr>
    <tr><td class="gt_row gt_right">34</td>
<td class="gt_row gt_right">18</td>
<td class="gt_row gt_right">13</td>
<td class="gt_row gt_right">1</td>
<td class="gt_row gt_right">34</td>
<td class="gt_row gt_left">rural</td>
<td class="gt_row gt_right">67.3</td></tr>
  </tbody>
  
  
</table>
</div>
</div>
<p class="optional-end">
</p>
<p><br></p>
<p>Recall the wellbeing/rurality study data? Let’s read them into R below:</p>
<pre class="r"><code>library(tidyverse)
mwdata &lt;- read_csv(&quot;https://uoepsy.github.io/data/wellbeing_rural.csv&quot;)
head(mwdata)</code></pre>
<pre><code>## # A tibble: 6 x 7
##     age outdoor_time social_int routine wellbeing location steps_k
##   &lt;dbl&gt;        &lt;dbl&gt;      &lt;dbl&gt;   &lt;dbl&gt;     &lt;dbl&gt; &lt;chr&gt;      &lt;dbl&gt;
## 1    28           12         13       1        36 rural       21.6
## 2    56            5         15       1        41 rural       12.3
## 3    25           19         11       1        35 rural       49.8
## 4    60           25         15       0        35 rural       NA  
## 5    19            9         18       1        32 rural       48.1
## 6    34           18         13       1        34 rural       67.3</code></pre>
<p>Suppose you are interested in whether an individual’s routine and amount of social interactions per week impacts their wellbeing above and beyond their age, location and time spent outdoors.</p>
<p>Routine and amount of social interactions per week are 2 predictors, while age, location and time spent outdoors are 4 predictors (location needs 2 dummy variables!). So we are comparing a smaller model <span class="math inline">\(m\)</span> with <span class="math inline">\(k = 4\)</span> predictors to a larger model <span class="math inline">\(M\)</span> with <span class="math inline">\(K = 4 + 2 = 6\)</span> predictors.</p>
<p>Past experience suggests that age, location and time spent outdoors account for roughly 22% of the variability in wellbeing.</p>
<pre class="r"><code>mdl_s &lt;- lm(wellbeing ~ age + outdoor_time + location, data = mwdata)
summary(mdl_s)</code></pre>
<pre><code>## 
## Call:
## lm(formula = wellbeing ~ age + outdoor_time + location, data = mwdata)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -12.2877  -3.0939  -0.2253   3.1799  18.9004 
## 
## Coefficients:
##                Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept)    35.92645    1.51457  23.721  &lt; 2e-16 ***
## age            -0.01104    0.02281  -0.484  0.62899    
## outdoor_time    0.17027    0.04794   3.551  0.00048 ***
## locationrural  -4.47646    0.79657  -5.620 6.54e-08 ***
## locationsuburb -0.16077    0.96689  -0.166  0.86811    
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 4.777 on 195 degrees of freedom
## Multiple R-squared:  0.231,  Adjusted R-squared:  0.2152 
## F-statistic: 14.64 on 4 and 195 DF,  p-value: 1.78e-10</code></pre>
<p>From a practical standpoint, it would be interesting if routine and social interactions per week accounted for at least 10% above this figure. Assuming a significance level of 0.05, how many subjects would be needed to identify such a contribution with 90% confidence?</p>
<p>The effect size here is:
<span class="math display">\[
f^2 = \frac{R^2_M - R^2_m}{1 - R^2_M} = \frac{0.32 - 0.22}{1 - 0.32} = 0.147
\]</span></p>
<p>Here,</p>
<ul>
<li><code>sig.level = 0.05</code>,</li>
<li><code>power = 0.90</code>,</li>
<li><code>u</code> is the number of <strong>extra</strong> predictors, so it will be <span class="math inline">\(K - k = 6 - 4 = 2\)</span></li>
<li>the effect size is the value computed before, <code>f2 = 0.147</code>.</li>
</ul>
<p>Entering this into the function yields the following:</p>
<pre class="r"><code>pwr.f2.test(u = 2, f2 = 0.147, sig.level = 0.05, power = 0.90)</code></pre>
<pre><code>## 
##      Multiple regression power calculation 
## 
##               u = 2
##               v = 86.15264
##              f2 = 0.147
##       sig.level = 0.05
##           power = 0.9</code></pre>
<p>In multiple regression, the denominator degrees of freedom <code>v</code> equals <span class="math inline">\(n - K - 1\)</span>, where <span class="math inline">\(n\)</span> is the sample size and <span class="math inline">\(K\)</span> is the number of predictors in the larger model.
In this case, <span class="math inline">\(n - 6 - 1 = 87\)</span>, which means the required sample size is <span class="math inline">\(n = 87 + 6 + 1 = 94\)</span>.</p>
</div>
<div id="power-curves" class="section level2">
<h2>Power curves</h2>
<p>It might be also interesting to investigate the joint relationship between sample size and effect size.</p>
<p>First, we create a tibble containing a sequence <code>seq</code> of effect sizes to investigate, ranging for example from 0.1 to 0.9 in steps of 0.01: <code>es = seq(.1, .9, .01)</code>.</p>
<p>Then, for each value in that sequence, we want to apply some computation. The computation is defined in</p>
<pre><code>map_dbl(es, 
        ~ pwr.f2.test(u = 2, f2 = .x, sig.level = 0.05, power = 0.9)$v)</code></pre>
<p>This takes each element of <code>es</code>, one at a time, and then passes it to the computation specified afterwards, into the slot <code>.x</code>. Then, function <code>pwr.f2.test</code> is applied and we want to extract the value of <code>v</code>.</p>
<pre class="r"><code>es_curve &lt;- tibble(
    es = seq(.1, .9, .01),
    v = map_dbl(es, 
                ~ pwr.f2.test(u = 2, f2 = .x, sig.level = 0.05, power = 0.9)$v)
)
es_curve</code></pre>
<pre><code>## # A tibble: 81 x 2
##       es     v
##    &lt;dbl&gt; &lt;dbl&gt;
##  1  0.1  127. 
##  2  0.11 115. 
##  3  0.12 106. 
##  4  0.13  97.4
##  5  0.14  90.5
##  6  0.15  84.4
##  7  0.16  79.2
##  8  0.17  74.5
##  9  0.18  70.4
## 10  0.19  66.7
## # … with 71 more rows</code></pre>
<p>Now, we want to compute <span class="math inline">\(n\)</span> from <code>v</code>, using <span class="math inline">\(n - K - 1 = \texttt v\)</span> so that <span class="math inline">\(n = \texttt v + K + 1\)</span>:</p>
<pre class="r"><code>es_curve &lt;- es_curve %&gt;%
    mutate(n = v + 6 + 1)
es_curve</code></pre>
<pre><code>## # A tibble: 81 x 3
##       es     v     n
##    &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
##  1  0.1  127.  134. 
##  2  0.11 115.  122. 
##  3  0.12 106.  113. 
##  4  0.13  97.4 104. 
##  5  0.14  90.5  97.5
##  6  0.15  84.4  91.4
##  7  0.16  79.2  86.2
##  8  0.17  74.5  81.5
##  9  0.18  70.4  77.4
## 10  0.19  66.7  73.7
## # … with 71 more rows</code></pre>
<p>Let’s see at the sample size required to find each effect size:</p>
<pre class="r"><code>ggplot(es_curve, aes(n, es)) +
    geom_line() +
    labs(x = &quot;Sample Size&quot;,
         y = &quot;Effect Size&quot;,
         title = &quot;Linear regression with power = 0.90 and alpha = 0.05&quot;)</code></pre>
<p><img src="17_power_regression_files/figure-html/unnamed-chunk-12-1.png" width="60%" style="display: block; margin: auto;" /></p>
<p>From the plot it looks like there is little gain in going beyond 100 observations.</p>
</div>
</div>
<div id="exercises" class="section level1">
<h1>Exercises</h1>
<p>A sample of 200 individuals aged 18-75 years old was randomly allocated to either take a free-recall test after a 1-hour long roller-coaster session (group = 0) or a meditation session (group = 1). The data available from at the following link: <a href="https://uoepsy.github.io/data/recall_med_coast.csv" class="uri">https://uoepsy.github.io/data/recall_med_coast.csv</a></p>
<ul>
<li><code>perc_recall</code>: the percentage of correctly recalled items</li>
<li><code>group</code>: whether the individual was assigned to the roller-coaster (group = 0) or the meditation (group = 1) session</li>
<li><code>age</code>: the individual’s age</li>
</ul>
<table class="table" style="width: auto !important; margin-left: auto; margin-right: auto;">
<thead>
<tr>
<th style="text-align:right;">
perc_recall
</th>
<th style="text-align:right;">
group
</th>
<th style="text-align:right;">
age
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:right;">
47.42891
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
52
</td>
</tr>
<tr>
<td style="text-align:right;">
61.44327
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
37
</td>
</tr>
<tr>
<td style="text-align:right;">
50.08974
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
46
</td>
</tr>
<tr>
<td style="text-align:right;">
56.44092
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
72
</td>
</tr>
<tr>
<td style="text-align:right;">
57.04785
</td>
<td style="text-align:right;">
1
</td>
<td style="text-align:right;">
46
</td>
</tr>
<tr>
<td style="text-align:right;">
46.11492
</td>
<td style="text-align:right;">
0
</td>
<td style="text-align:right;">
69
</td>
</tr>
</tbody>
</table>
<div class="question-begin">
Question 1
</div>
<div class="question-body">
<p>Using a significance level of 0.05, what sample size would you require to check whether any of the predictors (and interaction) influences recall scores with a 90% chance?</p>
<p>Because you do not know the effect size, assume Cohen’s guideline for linear regression and, to be on the safe side, consider the “Small” value.</p>
<p class="question-end">
</p>
</div>
<div class="solution-begin">
Solution <span id="sol-start-218" class="fa fa-plus solution-icon clickable" onclick="toggle_visibility(&#39;sol-body-218&#39;, &#39;sol-start-218&#39;)"></span>
</div>
<div id="sol-body-218" class="solution-body" style="display: none;">
<pre class="r"><code>k &lt;- 3
f2 &lt;- 0.02
pwr.f2.test(u = k, f2 = f2, sig.level = 0.05, power = 0.9)</code></pre>
<pre><code>## 
##      Multiple regression power calculation 
## 
##               u = 3
##               v = 708.495
##              f2 = 0.02
##       sig.level = 0.05
##           power = 0.9</code></pre>
<p>The required sample size is <span class="math inline">\(n = \texttt v + k + 1 = 709 + 3 + 1 = 713\)</span>.</p>
</div>
<p class="solution-end">
</p>
<div class="question-begin">
Question 2
</div>
<div class="question-body">
<p>Using the same <span class="math inline">\(\alpha\)</span> and power, what would be the sample size if you assumed effect size to be medium?</p>
<p class="question-end">
</p>
</div>
<div class="solution-begin">
Solution <span id="sol-start-219" class="fa fa-plus solution-icon clickable" onclick="toggle_visibility(&#39;sol-body-219&#39;, &#39;sol-start-219&#39;)"></span>
</div>
<div id="sol-body-219" class="solution-body" style="display: none;">
<pre class="r"><code>k &lt;- 3
f2 &lt;- 0.15
pwr.f2.test(u = k, f2 = f2, sig.level = 0.05, power = 0.9)</code></pre>
<pre><code>## 
##      Multiple regression power calculation 
## 
##               u = 3
##               v = 94.48157
##              f2 = 0.15
##       sig.level = 0.05
##           power = 0.9</code></pre>
<p>The required sample size is <span class="math inline">\(n = \texttt v + k + 1 = 95 + 3 + 1 = 99\)</span> or <span class="math inline">\(100\)</span> (to be on the safe side).</p>
</div>
<p class="solution-end">
</p>
<div class="question-begin">
Question 3
</div>
<div class="question-body">
<p>Using the same <span class="math inline">\(\alpha\)</span> and power, what would be the sample size if you assumed effect size to be large?</p>
<p class="question-end">
</p>
</div>
<div class="solution-begin">
Solution <span id="sol-start-220" class="fa fa-plus solution-icon clickable" onclick="toggle_visibility(&#39;sol-body-220&#39;, &#39;sol-start-220&#39;)"></span>
</div>
<div id="sol-body-220" class="solution-body" style="display: none;">
<pre class="r"><code>k &lt;- 3
f2 &lt;- 0.35
pwr.f2.test(u = k, f2 = f2, sig.level = 0.05, power = 0.9)</code></pre>
<pre><code>## 
##      Multiple regression power calculation 
## 
##               u = 3
##               v = 40.61744
##              f2 = 0.35
##       sig.level = 0.05
##           power = 0.9</code></pre>
<p>The required sample size is <span class="math inline">\(n = \texttt v + k + 1 = 41 + 3 + 1 = 45\)</span>.</p>
</div>
<p class="solution-end">
</p>
<div class="question-begin">
Question 4
</div>
<div class="question-body">
<p>Read the data into R and make sure that all variables are correctly encoded.</p>
<p class="question-end">
</p>
</div>
<div class="solution-begin">
Solution <span id="sol-start-221" class="fa fa-plus solution-icon clickable" onclick="toggle_visibility(&#39;sol-body-221&#39;, &#39;sol-start-221&#39;)"></span>
</div>
<div id="sol-body-221" class="solution-body" style="display: none;">
<pre class="r"><code>recdata &lt;- read_csv(&#39;https://uoepsy.github.io/data/recall_med_coast.csv&#39;)
head(recdata)</code></pre>
<pre><code>## # A tibble: 6 x 3
##   perc_recall group   age
##         &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
## 1        47.4     0    52
## 2        61.4     1    37
## 3        50.1     0    46
## 4        56.4     1    72
## 5        57.0     1    46
## 6        46.1     0    69</code></pre>
<p>Group should be a factor:</p>
<pre class="r"><code>recdata$group &lt;- factor(recdata$group, 
                        levels = c(0, 1), 
                        labels = c(&#39;rollercoaster&#39;, &#39;meditation&#39;))
head(recdata)</code></pre>
<pre><code>## # A tibble: 6 x 3
##   perc_recall group           age
##         &lt;dbl&gt; &lt;fct&gt;         &lt;dbl&gt;
## 1        47.4 rollercoaster    52
## 2        61.4 meditation       37
## 3        50.1 rollercoaster    46
## 4        56.4 meditation       72
## 5        57.0 meditation       46
## 6        46.1 rollercoaster    69</code></pre>
</div>
<p class="solution-end">
</p>
<div class="question-begin">
Question 5
</div>
<div class="question-body">
<p>Display a scatterplot investigating the relationship between <code>recall</code> and <code>age</code> by <code>group.</code>
Using <code>geom_smooth()</code>, also display the regression line for each group.</p>
<p>Is there any evidence of an interaction between age and group?</p>
<p class="question-end">
</p>
</div>
<div class="solution-begin">
Solution <span id="sol-start-222" class="fa fa-plus solution-icon clickable" onclick="toggle_visibility(&#39;sol-body-222&#39;, &#39;sol-start-222&#39;)"></span>
</div>
<div id="sol-body-222" class="solution-body" style="display: none;">
<pre class="r"><code>ggplot(recdata, aes(age, perc_recall, color = group)) +
    geom_point() +
    geom_smooth(method = lm, se = FALSE)</code></pre>
<p><img src="17_power_regression_files/figure-html/unnamed-chunk-20-1.png" width="60%" style="display: block; margin: auto;" /></p>
<p>The slope appears to be stepper in the roller coaster group than the meditation group, so we should investigate whether the interaction is significant.</p>
</div>
<p class="solution-end">
</p>
<div class="question-begin">
Question 6
</div>
<div class="question-body">
<p>Fit a model with just age as a predictor of recall.
What is the R-squared?</p>
<p class="question-end">
</p>
</div>
<div class="solution-begin">
Solution <span id="sol-start-223" class="fa fa-plus solution-icon clickable" onclick="toggle_visibility(&#39;sol-body-223&#39;, &#39;sol-start-223&#39;)"></span>
</div>
<div id="sol-body-223" class="solution-body" style="display: none;">
<pre class="r"><code>s1 &lt;- summary(lm(perc_recall ~ age, data = recdata))
s1</code></pre>
<pre><code>## 
## Call:
## lm(formula = perc_recall ~ age, data = recdata)
## 
## Residuals:
##      Min       1Q   Median       3Q      Max 
## -10.0850  -4.7474   0.7331   4.6758  10.1733 
## 
## Coefficients:
##             Estimate Std. Error t value Pr(&gt;|t|)    
## (Intercept) 62.16336    1.81984  34.159  &lt; 2e-16 ***
## age         -0.16206    0.03672  -4.414 2.61e-05 ***
## ---
## Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1
## 
## Residual standard error: 5.494 on 98 degrees of freedom
## Multiple R-squared:  0.1658, Adjusted R-squared:  0.1573 
## F-statistic: 19.48 on 1 and 98 DF,  p-value: 2.614e-05</code></pre>
<pre class="r"><code>s1$r.squared</code></pre>
<pre><code>## [1] 0.1658256</code></pre>
<pre class="r"><code>s1$adj.r.squared</code></pre>
<pre><code>## [1] 0.1573136</code></pre>
<p>We can use either, for multiple linear regression, you might want to use the adjusted R-squared: <span class="math inline">\(R^2 = 0.16\)</span>.</p>
</div>
<p class="solution-end">
</p>
<div class="question-begin">
Question 7
</div>
<div class="question-body">
<p>Imagine you found the R-squared that you computed above in a paper, and you are using that to base your next study.</p>
<p>A researcher believes that the inclusion of group and its interaction with age should explain an extra 50% of the variation in recall scores.</p>
<p>Using a significance level of 5%, what sample size should you use for your next data collection in order to discover that effect with a power of 0.9?</p>
<p class="question-end">
</p>
</div>
<div class="solution-begin">
Solution <span id="sol-start-224" class="fa fa-plus solution-icon clickable" onclick="toggle_visibility(&#39;sol-body-224&#39;, &#39;sol-start-224&#39;)"></span>
</div>
<div id="sol-body-224" class="solution-body" style="display: none;">
<pre class="r"><code># smaller model m
k &lt;- 1
R2m &lt;- 0.16

# larger model M
K &lt;- 3
R2M &lt;- 0.16 + 0.5

# effect size
f2 &lt;- (R2M - R2m) / (1 - R2M)
pwr.f2.test(u = K - k, f2 = f2, sig.level = 0.05, power = 0.9)</code></pre>
<pre><code>## 
##      Multiple regression power calculation 
## 
##               u = 2
##               v = 9.211914
##              f2 = 1.470588
##       sig.level = 0.05
##           power = 0.9</code></pre>
<p>The sample size should be <span class="math inline">\(n = \texttt v + K + 1 = 10 + 3 + 1 = 14\)</span>.</p>
<p>With such a big effect size, don’t be surprised it’s so small. When the effect size is much smaller, that will be harder to detect and you will require a bigger sample size.</p>
</div>
<p class="solution-end">
</p>
<div class="question-begin">
Question 8
</div>
<div class="question-body">
<p>Suppose that the addition of group and its interaction with age, to the R-squared from the literature, was thought to explain only 5% more of the variability in recall scores.</p>
<p>Using a significance level of 5%, what sample size should you use for your next data collection in order to discover that effect with a power of 0.9?</p>
<p class="question-end">
</p>
</div>
<div class="solution-begin">
Solution <span id="sol-start-225" class="fa fa-plus solution-icon clickable" onclick="toggle_visibility(&#39;sol-body-225&#39;, &#39;sol-start-225&#39;)"></span>
</div>
<div id="sol-body-225" class="solution-body" style="display: none;">
<pre class="r"><code># smaller model m
k &lt;- 1
R2m &lt;- 0.16

# larger model M
K &lt;- 3
R2M &lt;- 0.16 + 0.05

# effect size
f2 &lt;- (R2M - R2m) / (1 - R2M)
pwr.f2.test(u = K - k, f2 = f2, sig.level = 0.05, power = 0.9)</code></pre>
<pre><code>## 
##      Multiple regression power calculation 
## 
##               u = 2
##               v = 199.9608
##              f2 = 0.06329114
##       sig.level = 0.05
##           power = 0.9</code></pre>
<p>The sample size should be <span class="math inline">\(n = \texttt v + K + 1 = 200 + 3 + 1 = 124\)</span>.</p>
<p>With such a small effect size, we need a bigger sample size for us to detect it with high confidence.</p>
</div>
<p class="solution-end">
</p>
<!-- Formatting -->
<div class="tocify-extend-page" data-unique="tocify-extend-page" style="height: 0;">

</div>
</div>
<div class="footnotes">
<hr />
<ol>
<li id="fn1"><p>Cohen, J. (1988) <em>Statistical Power Analysis for the Behavioral Sciences</em>, 2nd ed. Hillsdale, NJ: Lawrence Erlbaum.<a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>Cohen J. (1992) A power primer, <em>Psychological Bulletin, 112</em>(1): 155–159. PDF available via the University of Edinburgh library at <a href="https://discovered.ed.ac.uk/permalink/f/1s15qcp/TN_cdi_proquest_journals_614317877">this link.</a><a href="#fnref2" class="footnote-back">↩︎</a></p></li>
</ol>
</div>

<link rel="stylesheet" href="https://uoepsy.github.io/assets/css/ccfooter.css" />
<div class="ccfooter"></div>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
